Final Pull Request from DA to DA_PullReqs

TMaile opened the pull request:
This includes the final code from us. It also contains earlier pull requests between other DA branches.

TMaile committed:
Merge remote-tracking branch 'origin/DA' into DA

mrigeshgupta committed:
create json data from reading weather file.

TMaile committed:
adding option to skip schema validation

mrigeshgupta committed:
convert workflow maker code into json object

mrigeshgupta committed:
convert workflow maker code into json object

mrigeshgupta committed:
convert workflow maker code into json object

TMaile committed:
update on JSON driven measure selection

mrigeshgupta committed:
convert workflow maker code into json object

mrigeshgupta committed:
add new properties

TMaile committed:
added option to remove whitespace from space names

mrigeshgupta committed:
some bug fixing in downloading weather file

TMaile committed:
fixing issue with space name

TMaile committed:
Merge remote-tracking branch 'origin/DA' into DA

TMaile committed:
fixed typo

TMaile committed:
improving logging of added measures

mrigeshgupta committed:
new property has been created for mapping spreadsheet.

TMaile committed:
adjustments for the reporting measure

TMaile committed:
small adjustment

TMaile committed:
Merge remote-tracking branch 'remotes/origin/DA_Level1' into DA

TMaile committed:
Merge remote-tracking branch 'remotes/origin/DA' into DA_Level1

TMaile committed:
Merge branch 'DA_Level1' into DA

mrigeshgupta committed:
Change SubSection into Section, because 2.0 only support section.

mrigeshgupta committed:
Merge branch 'DA' into DA_Level1

mrigeshgupta committed:
Merge branch 'DA_Level1' into DA_Level1_BldgSync2.0

TMaile committed:
adding a test file

TMaile committed:
Merge remote-tracking branch 'origin/DA_Level1_BldgSync2.0' into DA_Level1_BldgSync2.0

# Conflicts:
#	spec/files/building_151.xml

TMaile committed:
adding tests for new building level 1 parameters

mrigeshgupta committed:
Change to support EDV project

mrigeshgupta committed:
bug fixing

mrigeshgupta committed:
stop selection tool spec file so travis file doesn't show error.

TMaile committed:
adding tests for new building level 1 parameters

TMaile committed:
adding tests for new building level 1 parameters - Section

TMaile committed:
adding tests for new building level 1 parameters - hvac system

TMaile committed:
updating travis file for updated file name

TMaile committed:
fixing issue with hvac system init

mrigeshgupta committed:
create json file for spatial element class

TMaile committed:
processing level 1 beta properties, occupacy schedules

TMaile committed:
Merge remote-tracking branch 'origin/DA_Level1' into DA_Level1

TMaile committed:
added a nil check

TMaile committed:
fixed issue with office type

TMaile committed:
fixed issue with office type

mrigeshgupta committed:
implement json in spatial element class object

mrigeshgupta committed:
implement json in spatial element class object

mrigeshgupta committed:
changes in workflow maker class to create xml file.

TMaile committed:
going back to subsection

TMaile committed:
Merge remote-tracking branch 'origin/DA_Level1' into DA_Level1

mrigeshgupta committed:
merge changes from DA_level1

TMaile committed:
going back to subsection

mrigeshgupta committed:
Variable move into OpenStudio-Extension project

mrigeshgupta committed:
Merge branch 'DA' into DA_Level1

# Conflicts:
#	lib/buildingsync/makers/workflow_maker_phase_zero.rb

TMaile committed:
integrating create parametric schedule into workflow

TMaile committed:
Merge remote-tracking branch 'origin/DA_Level1' into DA_Level1

TMaile committed:
checking if there are any building sections

TMaile committed:
rename classes and files from subsection to section

TMaile committed:
Merge remote-tracking branch 'origin/DA_Level1_BldgSync2.0' into DA_Level1_BldgSync2.0

TMaile committed:
more cleanup subsection versus section

TMaile committed:
update file for sections

TMaile committed:
adding new function to clear all existing measures

TMaile committed:
Merge remote-tracking branch 'origin/DA' into DA

TMaile committed:
adding the adjusted test file

TMaile committed:
fixing issue with empty workflow steps

TMaile committed:
Merge remote-tracking branch 'remotes/origin/DA' into DA_Level1

TMaile committed:
Merge remote-tracking branch 'remotes/origin/DA_Level1' into DA_Level1_BldgSync2.0

# Conflicts:
#	lib/buildingsync/makers/workflow_maker_phase_zero.rb

TMaile committed:
fixing a merge bug

TMaile committed:
fixing a merge bug

TMaile committed:
SIMULATE_BASELINE_ONLY, DO_MODEL_CALIBRATION, and DO_GET_RESULTS constants added to BldgSync, cleanup of config.rb.in

TMaile committed:
looks like merge leftovers

TMaile committed:
fixed some issues around results gathering

TMaile committed:
removing some put statements

TMaile committed:
adding some put statements

TMaile committed:
adding some put statements

TMaile committed:
adding new error message for climate zone issues

TMaile committed:
adjusting the logging

TMaile committed:
adjusting the logging

TMaile committed:
fixing issue when stat file does not exist

TMaile committed:
adding more test files

TMaile committed:
call parametric schedule measure directly and adjust testing

TMaile committed:
adding new test file

TMaile committed:
adding monthly data

TMaile committed:
adjust system type

TMaile committed:
update to enable gather results on baseline run

TMaile committed:
adding another test file

TMaile committed:
adding new test file

TMaile committed:
adding new test file

TMaile committed:
adding the ASHRAE Level 1 use case, adding the error if we find 2 buildings

TMaile committed:
catching exceptions while gathering results

TMaile committed:
skipping measure scenario

TMaile committed:
adding ddy files as parameter

TMaile committed:
more ddy file changes

TMaile committed:
Merge branch 'DA' into DA_Level1

TMaile committed:
Merge branch 'DA_Level1' into DA_Level1_BldgSync2.0

TMaile committed:
adding ddy file as optional parameter

TMaile committed:
Merge remote-tracking branch 'origin/DA_Level1_BldgSync2.0' into DA_Level1_BldgSync2.0

TMaile committed:
Merge pull request #39 from BuildingSync/DA_Level1

milestone level 1 beta

TMaile committed:
fixed syntax issue

TMaile committed:
updating test files

TMaile committed:
updating the Golden Test file test

TMaile committed:
Merge pull request #40 from BuildingSync/DA_Level1_BldgSync2.0

update to BuildingSync version 2.0 prerelease

TMaile committed:
Merge pull request #41 from BuildingSync/DA_Level1

Da level1

TMaile committed:
bug fixing to handle sections in the new way

TMaile committed:
adding the new test file

TMaile committed:
fixing bug with old sections

TMaile committed:
fix issue with building count, we first check the count before processing further

TMaile committed:
remove temporary workaround to remove whitespaces

TMaile committed:
updated branch for openstudio-extensio-gem, since we did merge our changes over there

TMaile committed:
updated test

TMaile committed:
updated test

TMaile committed:
fix issue with link to incorrect branch

Commit:
needed openstudio/extension for defining constants in edv process_all_bldg...

TMaile committed:
adding the extension require

TMaile committed:
fixed typo in SIMULATE_BASELINE_ONLY

TMaile committed:
Merge pull request #42 from BuildingSync/DA_Level1_BldgSync2.0

Da level1 bldg sync2.0

TMaile committed:
removing lodging and laboratory testing as occupancy types

TMaile committed:
Merge remote-tracking branch 'origin/DA' into DA

TMaile committed:
adding tests for occupancy type

TMaile committed:
adding tests for all occupancy types

TMaile committed:
fixing test issues and moving the occupany type testing to a new file since it takes too long for Travis

TMaile committed:
adding the moved occupany type testing

TMaile committed:
Merge remote-tracking branch 'remotes/origin/DA' into DA_Test

# Conflicts:
#	lib/buildingsync/makers/model_maker_level_zero.rb
#	lib/buildingsync/model_articulation/spatial_element.rb

TMaile committed:
Merge branch 'DA_Test' into DA_Level1_BldgSync2.0

TMaile committed:
Merge pull request #43 from BuildingSync/DA_Level1_BldgSync2.0

Sync DA Test with DA Level 1 BldgSync 2.0

TMaile committed:
Merge branch 'DA_Test' of https://github.com/BuildingSync/BuildingSync-gem into DA_Test

TMaile committed:
adding footprint

TMaile committed:
adding principal hvac and lighting types

TMaile committed:
refactor the set method

TMaile committed:
adding more system types

TMaile committed:
add the set logic to site

TMaile committed:
fixing errors

TMaile committed:
adding the moved occupany type testing

TMaile committed:
enabling baseline run with new runner constants

TMaile committed:
more adjustments for the runner changes

TMaile committed:
completing "Contact Information and Audit Details" level 1.5 parameters

TMaile committed:
more changes related to level 1.5 attributes

Commit:
bump version for openstudio-model articulation gem in gemspec

Commit:
Merge remote-tracking branch 'origin/develop' into DA

Commit:
update dates in templates. Add changelog and license. Update gemfile to use allow_local.

Commit:
remove unneeded fields

Commit:
fix tests by using checkout of model articulation DA branch

Commit:
rubocop cleanup

Commit:
test os 2.8.1 and 2.9.0

TMaile committed:
changing branch of openstudio-extension

TMaile committed:
Merge remote-tracking branch 'remotes/origin/cleanup-files-for-release' into DA_Test

# Conflicts:
#	lib/buildingsync/makers/workflow_maker_phase_zero.rb

TMaile committed:
adding more parameters and the lighting system

TMaile committed:
fixing typos and adding new testfile

TobiasMaile committed:
fixing failures in facility_spec

Commit:
Merge branch 'DA_Test' of https://github.com/BuildingSync/BuildingSync-gem into DA_Test

Merge remote to local

TobiasMaile committed:
fixing issue with determining the floor area

TobiasMaile committed:
fixing issue with occupancy classification, resolves #46

TobiasMaile committed:
fixing issues with total floor area and occupancy type selection

TobiasMaile committed:
fixing issue with occupancy classification, resolves #46

TobiasMaile committed:
fixing the last test issue

TMaile committed:
Merge branch 'DA' into cleanup-files-for-release

TobiasMaile committed:
fixing issues with reports report

TobiasMaile committed:
fixing test issues

TobiasMaile committed:
skipping L100.xml for now

TMaile committed:
Merge branch 'DA_Test' into DA

TMaile committed:
Merge pull request #50 from BuildingSync/DA

Forward merge code to DA_Test

TobiasMaile committed:
update the spec helper

TobiasMaile committed:
refactor the model maker names and fixing a test failure

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

TobiasMaile committed:
more cleanup of model and workflow maker

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

Commit:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

Commit:
rename workflow_maker to workflow_maker_base

Commit:
gather_results_and_save_xml, remove save_xml

Commit:
undo previous, didnt see it had already been done...

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

# Conflicts:
#	lib/buildingsync/translator.rb

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

TobiasMaile committed:
more cleanup of model and other code

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

TobiasMaile committed:
separate workflow maker better from model maker

TobiasMaile committed:
adding function to get at scenarios

TobiasMaile committed:
cleanup testing code

TobiasMaile committed:
used a different report measure for testing that is always available

TobiasMaile committed:
checking if bundle -v works on installation

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

TobiasMaile committed:
getting the linked building

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

TobiasMaile committed:
getting the linked building

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

TobiasMaile committed:
adding more testcases cleanup of testruns

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

TobiasMaile committed:
fixing issues with floor areas

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

mrigeshgupta committed:
Initialize the reverse engineering in parameter to xml file.

mrigeshgupta committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

# Conflicts:
#	lib/buildingsync/model_articulation/spatial_element.rb

mrigeshgupta committed:
fixing the null exception

TMaile committed:
Merge branch 'DA' into cleanup-files-for-release

TobiasMaile committed:
bugs fixes

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

TobiasMaile committed:
fixing the geometry issue with factions of multiple sections

TobiasMaile committed:
adding the HVAC mapping

TobiasMaile committed:
bugs fixes ends

TobiasMaile committed:
adding runner options

TobiasMaile committed:
finally getting the multiple HVAC systems to work

TobiasMaile committed:
fix bug with no systems in the xml file

Commit:
update save_xml call in gather_results_and_save_xml

Commit:
edit getting list of zones by space_type

TobiasMaile committed:
fix bug with no sections and new support for multiple hvac systems

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

# Conflicts:
#	lib/buildingsync/model_articulation/building.rb

TobiasMaile committed:
fix testing failures due to changes in the hvac sytem class

TobiasMaile committed:
fixing issue with occupancy adjustment, moving L000 and L001 tests

TobiasMaile committed:
fixing issue with scenario name and adding the office_carolina test file

Commit:
fix pathing for runner

Commit:
Merge branch 'DA_Test' of https://github.com/BuildingSync/BuildingSync-gem into DA_Test

Merge remote to local

Commit:
add case 2.3 for @city and @state defined at the building level

TobiasMaile committed:
fixing issue with occupancy correction

TobiasMaile committed:
Merge remote-tracking branch 'origin/DA_Test' into DA_Test

TMaile committed:
Merge pull request #44 from BuildingSync/cleanup-files-for-release

Cleanup files needed for release

TobiasMaile committed:
improving the occupancy adjustment per section now

TobiasMaile committed:
Merge branch 'DA' into DA_Test

TMaile committed:
Merge branch 'DA_Test' into DA

# Conflicts:
#	lib/buildingsync/makers/workflow_maker_phase_zero.rb

TMaile committed:
Fix for gem file and simulation testing function

TMaile committed:
Fix for gem file and simulation testing function

nllong commented:
@corymosiman12 @JanghyunJK -- any reason to not merge this down?

corymosiman12 commented on config.rb.in:
```diff
@@ -21,30 +21,6 @@ module BuildingSync
         File.realpath("./../openstudio-measures/nrel_dev", File.dirname(__FILE__))
     ]
 
-  # todo: move those constants into the extension-gem
-
-  # max number of datapoints to run
-  MAX_DATAPOINTS = Float::INFINITY
-  #MAX_DATAPOINTS = 2
-
-  # number of parallel jobs per BuildingSync file
-  NUM_PARALLEL = 4
-
   # number of parallel BuildingSync files to run
   NUM_BUILDINGS_PARALLEL = 2
-
-  # do simulations
-  DO_SIMULATIONS = false
-  
-  # specify the output folder for the simulation runs
-  #SIMULATION_OUTPUT_FOLDER = "D:/BRICR_OUTPUT/"
-
-  # specify to run the baseline simulation only or not
-  SIMULATE_BASELINE_ONLY = false
-
-  # specify to include the model calibration or not
-  DO_MODEL_CALIBRATION = false
-  
-  # collect results
-  DO_GET_RESULTS = false
```
Extension gem defines the following as options for [runner](https://github.com/NREL/openstudio-extension-gem/blob/cbcbe47008d42e7c72dd0f897cf59351bc03e17a/lib/openstudio/extension/runner.rb#L62-L64)
- max_datapoints ~ MAX_DATAPOINTS
- num_parallel ~ NUM_PARALLEL
- run_simulations ~ DO_SIMULATIONS

Does @nllong @TMaile  know what is meant by `max_datapoints` - can we add that definition into Runner?

They BSync-gem defines the following as (global?) variables in [extension](https://github.com/BuildingSync/BuildingSync-gem/blob/bb52655ebb8efeca44249277d3fb67ac60b4e610/lib/buildingsync/extension.rb#L42-L51)

- SIMULATE_BASELINE_ONLY
- DO_MODEL_CALIBRATION
- DO_GET_RESULTS
- NUM_BUILDINGS_PARALLEL

@nllong do we want to mimic the `Runner` style and pass these as options to `Translator` instead of as global variables?  Or how would you like this to look?

nllong replied:
part 1: It is defined on develop  - https://github.com/NREL/openstudio-extension-gem/blob/develop/lib/openstudio/extension/runner.rb#L63. 
part 2: yes, but I don't think we need to do that at this moment.

TMaile replied:
created a new issue to address this later #62

corymosiman12 commented on lib/buildingsync/get_bcl_weather_file.rb:
```diff
@@ -32,68 +32,82 @@
-      responses = remote.searchComponentLibrary(city, 'Weather File')
-      choices = OpenStudio::StringVector.new
+      if !weather_file_name.empty?
+        return File.expand_path("../../spec/weather/#{weather_file_name}", File.dirname(__FILE__))
```
@nllong do we want to store these files under `spec/`? Should we have a top level `data/` dir or something? Or good for now?

nllong replied:
if these are just for running tests, then in spec makes sense. Otherwise, yes we should move to a `lib/data` directory.

TMaile replied:
this is fixed now

corymosiman12 commented on lib/buildingsync/get_bcl_weather_file.rb:
```diff
@@ -187,10 +204,112 @@ def create_ddy_file(idf_path_collection, epw_path)
+      location = weather_file.readlines.first.split(',')
+
+      weather_file_name = File.basename(epw_path)
+      city_name = location[1]
+      state_code = location[2]
+      weather_id = location[5]
```
Do we know this always holds true for all weather files?

nllong replied:
probably 98%. As long as we error out nicely, then I say we are good.

TMaile replied:
added a check to ensure that if it is not true we log an error properly

corymosiman12 commented on lib/buildingsync/get_bcl_weather_file.rb:
```diff
@@ -187,10 +204,112 @@ def create_ddy_file(idf_path_collection, epw_path)
+      weather_json[:state_code] << state_code
+      weather_json[:weather_id] << weather_id
+
+      File.open(weather_file_path, 'w') { |f| f.write(weather_json.to_json) }
```
what is the purpose of the `weather.json` file?  I see the `read_json_file` method below, but doesn't appear to be used anywhere in the gem.  Can we quick document the expected use case for these functions?

corymosiman12 replied:
I see - it is a cached value, primarily used as a first pass before calling out to BCL via [get_weather_json_data](https://github.com/BuildingSync/BuildingSync-gem/blob/bb52655ebb8efeca44249277d3fb67ac60b4e610/lib/buildingsync/get_bcl_weather_file.rb#L291)

TMaile replied:
this JSON file is used to store EPW files that were already downloaded and thus do not need to be downloaded again.

corymosiman12 commented on doc_templates/copyright_erb.txt:
```diff
@@ -1,6 +1,6 @@
 <%
   # *******************************************************************************
-  # OpenStudio(R), Copyright (c) 2008-2018, Alliance for Sustainable Energy, LLC.
+  # OpenStudio(R), Copyright (c) 2008-2019, Alliance for Sustainable Energy, LLC.
```
update

TMaile replied:
done

corymosiman12 commented on doc_templates/copyright_js.txt:
```diff
@@ -1,5 +1,5 @@
 /* @preserve
- * OpenStudio(R), Copyright (c) 2008-2018, Alliance for Sustainable Energy, LLC. All rights reserved.
+ * OpenStudio(R), Copyright (c) 2008-2019, Alliance for Sustainable Energy, LLC. All rights reserved.
```
update

corymosiman12 commented on lib/buildingsync/makers/model_maker.rb:
```diff
@@ -56,17 +58,38 @@ def generate_baseline(dir, epw_file_path, standard_to_be_used)
         OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.ModelMakerLevelZero.generate_baseline', "There are more than one (#{@facilities.count})facilities in your BuildingSync file. Only one if supported right now")
```
Change error to `.....ModelMaker.read_xml....`

corymosiman12 replied:
@TMaile I think get rid of references to 'LevelZero':
https://github.com/BuildingSync/BuildingSync-gem/blob/7e8a02fcdc84797a8b2e9e1dd3e60d523bde6f17/lib/buildingsync/makers/model_maker.rb#L55
https://github.com/BuildingSync/BuildingSync-gem/blob/7e8a02fcdc84797a8b2e9e1dd3e60d523bde6f17/lib/buildingsync/makers/model_maker.rb#L58

TMaile replied:
resolved in https://github.com/BuildingSync/BuildingSync-gem/commit/f007a33cd39de5a14e8dfd779050f0e53ee8bdf2

corymosiman12 commented on lib/buildingsync/makers/model_maker.rb:
```diff
@@ -56,17 +58,38 @@ def generate_baseline(dir, epw_file_path, standard_to_be_used)
       end
+    end
+
+    def get_facility
+      return @facility
+    end
```
Change to `attr_accessor` or `attr_reader`?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      return measure_type
+    end
+
+    def get_workflow
```
`attr_reader` or?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+          measure_name = measure.elements["#{@ns}:TechnologyCategories/#{@ns}:TechnologyCategory/#{@ns}:BoilerPlantImprovements/#{@ns}:MeasureName"].text
+        end
+      elsif measure_category == 'Other HVAC'
+        measure_name = measure.elements["#{@ns}:TechnologyCategories/#{@ns}:TechnologyCategory/#{@ns}:*/#{@ns}:MeasureName"].text
```
why use `*` instead of `OtherHVAC`?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      elsif measure_category == 'Other HVAC'
+        measure_name = measure.elements["#{@ns}:TechnologyCategories/#{@ns}:TechnologyCategory/#{@ns}:*/#{@ns}:MeasureName"].text
+
+        # DLM: somme measures don't have a direct BuildingSync equivalent, use UserDefinedField 'OpenStudioMeasureName' for now
```
Need to research which measures don't have a BSync equivalent and document. add to UDF quashing task list?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+          end
+        end
+        if @scenarios.empty?
+          puts 'No scenarios found in your BuildingSync XML file!'
```
should all `puts` be changed to some sort of `log`?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      scenarios.each do |scenario|
+        scenario_name = scenario.elements["#{@ns}:ScenarioName"].text
+        puts "scenario with name #{scenario_name} found"
+        if scenario_name == 'Baseline'
```
need to change this to be in line with our definition for a baseline modeled scenario once we get this all into place.  Don't want to rely on the `@ID` or `auc:ScenarioName` text.  This type of scenario should be declared via:

`auc:Scenario/auc:ScenarioType/auc:CurrentBuilding/auc:CalculationMethod/auc:Modeled`

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+          scenario_element.add_element(scenario_type_element)
+
+          get_scenarios.add_element(scenario_element)
+          puts '.....adding a new baseline scenario'
```
^^ ditto

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      # write an osw for each scenario
+      scenarios.each do |scenario|
+        # get information about the scenario
+        scenario_name = scenario.elements["#{@ns}:ScenarioName"].text
```
Same thing for `Measured`:

`auc:Scenario/auc:ScenarioType/auc:CurrentBuilding/auc:CalculationMethod/auc:Measured`

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      end
+    end
+
+    def get_measure_result(result, measure_dir_name, result_name)
```
Where does this data currently get stored?  method doesn't seem to be called anywhere?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      return @failed_scenarios
+    end
+
+    def saveXML(filename)
```
change to `save_xml`

TMaile replied:
done in https://github.com/BuildingSync/BuildingSync-gem/commit/83a2dfbe1364d4c56414054d8d1753c4fbd86267

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+
+          puts "processing scenario with name #{scenario_name}"
+          # dir for the osw
+          osw_dir = File.join(dir, scenario_name)
+          # cleanup large files
+          path = File.join(osw_dir, 'eplusout.sql')
+          FileUtils.rm_f(path) if File.exist?(path)
+          path = File.join(osw_dir, 'data_point.zip')
+          FileUtils.rm_f(path) if File.exist?(path)
+          path = File.join(osw_dir, 'eplusout.eso')
+          FileUtils.rm_f(path) if File.exist?(path)
+          Dir.glob(File.join(osw_dir, '*create_typical_building_from_model*')).each do |path|
+            FileUtils.rm_rf(path) if File.exist?(path)
+          end
+          Dir.glob(File.join(osw_dir, '*create_typical_building_from_model*')).each do |path|
+            FileUtils.rm_rf(path) if File.exist?(path)
+          end
```
let's move large file cleanup to separate method

TMaile replied:
resolved through: https://github.com/BuildingSync/BuildingSync-gem/commit/7e8a02fcdc84797a8b2e9e1dd3e60d523bde6f17

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+            weather_data_type.text = 'TMY3'
+            modeled.add_element(weather_data_type)
+            sim_completion_status = REXML::Element.new("#{@ns}:SimulationCompletionStatus")
+            sim_completion_status.text = result[:completed_status] == 'Success' ? 'Finished' : 'Failed' # TODO: double check what these keys can be
```
Remove TODO.  Valid keys are:
- `Not Started`
- `Started`
- `Finished`
- `Failed`
- `Unknown`

TMaile replied:
removed todo in https://github.com/BuildingSync/BuildingSync-gem/commit/f0c9195fd1b3bbe52cc00bc417cb0eabab844950

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+
+    # DLM: total hack because these are not reported in the out.osw
+    # output is array of [source_energy, source_eui] in kBtu and kBtu/ft2
+    def getSourceEnergyArray(eplustbl_path)
```
rename: `get_source_energy_array`

TMaile replied:
changed in https://github.com/BuildingSync/BuildingSync-gem/commit/c9e5c3e85a76535eee888a3bcf3641b278012cb1

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      return result
+    end
+
+    def getMeasureResult(result, measure_dir_name, result_name)
```
rename: `get_measure_result`

TMaile replied:
resolved in https://github.com/BuildingSync/BuildingSync-gem/commit/aa8c332ae65e214c0cbcff8233e96ca7c88fa566

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+            package_of_measures.add_element(annual_savings_energy_cost)
+  
+            # KAF: adding annual savings by fuel
+            electricity_savings = baseline_fuel_electricity_kbtu - fuel_electricity_kbtu
```
add `.nil?` checking in case elec or natural gas is not used?

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+      end
+    end
+
+    def gather_results(dir, baseline_only = false)
```
in general I think we can break this method up into smaller functions / methods.  It is difficult to follow and unclear that:
1. Only electricity (both energy and peak demand) and NG resources are found
1. Only annual and monthly data is found / added

For example:
- `get_results_by_resource(fuel_type)`
- `gather_monthly_values_by_resource(fuel_type)`
...etc.

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+              timeseries.add_element(ts_quantity)
+              start_time = REXML::Element.new("#{@ns}:StartTimeStamp")
+              if month < 10
+                start_time.text = '2017-0' + month.to_s + '-01T00:00:00'
```
Year of timestamp should be set depending on the:
- If AMY data file used, should depend on that
- If TMY data file used, should maybe just use current year? Or should have some strategy.  Shouldn't just default to `2017` and `2018`

corymosiman12 commented on lib/buildingsync/makers/workflow_maker.rb:
```diff
@@ -0,0 +1,852 @@
+              timeseries.add_element(ts_quantity)
+              start_time = REXML::Element.new("#{@ns}:StartTimeStamp")
+              if month < 10
+                start_time.text = '2017-0' + month.to_s + '-01T00:00:00'
+              else
+                start_time.text = '2017-' + month.to_s + '-01T00:00:00'
+              end
+              timeseries.add_element(start_time)
+              end_time = REXML::Element.new("#{@ns}:EndTimeStamp")
+              if month < 9
+                end_time.text = '2017-0' + (month + 1).to_s + '-01T00:00:00'
+              elsif month < 12
+                end_time.text = '2017-' + (month + 1).to_s + '-01T00:00:00'
+              else
+                end_time.text = '2018-01-01T00:00:00'
```
^^ ditto

corymosiman12 commented on lib/buildingsync/model_articulation/building_section.rb:
```diff
@@ -0,0 +1,244 @@
+
+    def write_parameters_to_xml(ns, buildingSection)
+      buildingSection.elements["#{ns}:fraction_area"].text = @fraction_area
+      buildingSection.elements["#{ns}:UserDefinedFields/#{ns}:UserDefinedField/#{ns}:FieldValue"].text = @occupancy_classification_original if !@occupancy_classification_original.nil?
```
update in accordance with: https://github.com/BuildingSync/schema/pull/181

corymosiman12 replied:
In [v2.0.1](https://github.com/BuildingSync/schema/releases/tag/v2.1.0), this has been added as an element.  please update accordingly.

TMaile replied:
This is addressed in #76

corymosiman12 commented on lib/buildingsync/model_articulation/building_section.rb:
```diff
@@ -0,0 +1,244 @@
+    def write_parameters_to_xml(ns, buildingSection)
+      buildingSection.elements["#{ns}:fraction_area"].text = @fraction_area
+      buildingSection.elements["#{ns}:UserDefinedFields/#{ns}:UserDefinedField/#{ns}:FieldValue"].text = @occupancy_classification_original if !@occupancy_classification_original.nil?
+      buildingSection.elements["#{ns}:UserDefinedFields/#{ns}:UserDefinedField/#{ns}:FieldValue"].text = @principal_hvac_type if !@principal_hvac_type.nil?
```
An element now exists for this: https://github.com/BuildingSync/schema/blob/dc5da7d132c51f0456bb8c66e1946afe5a945828/BuildingSync.xsd#L6779

This should not be declared as a UDF anymore.  Same thing for next two lines.  See definitional requirements here: https://github.com/BuildingSync/TestSuite/blob/L100_Schematron/docs/Systems.md#l100-primary-systems

TMaile replied:
@corymosiman12 so you mean occupancy_classification_original and principal_hvac_type , correct?

corymosiman12 replied:
Maybe it is `auc:PrincipalHVACSystemType`

corymosiman12 replied:
but yes

TMaile replied:
This is addressed in #76

corymosiman12 commented on lib/buildingsync/model_articulation/foundation_system_type.rb:
```diff
@@ -0,0 +1,61 @@
+    end
+
+    def read(section_element, ns)
+      #if section_element.elements["#{ns}:FenestrationType/#{ns}:Door"]
```
need to add `pass` or equivalent statement here so doesn't fail? (not sure what ruby syntax is, maybe `next`)?  equivalent to python's `pass`

nllong replied:
I think this is okay. Just return nothing.

corymosiman12 commented on lib/buildingsync/model_articulation/loads_system.rb:
```diff
@@ -87,6 +92,119 @@ def add_internal_loads(model, standard, template, remove_objects)
+      end
+    end
+
+    def adjust_people_schedule_old(space_type, building_section, model)
```
Should this be removed if 'old'?

nllong replied:
yes

corymosiman12 commented on lib/buildingsync/model_articulation/loads_system.rb:
```diff
@@ -69,6 +70,10 @@ def add_internal_loads(model, standard, template, remove_objects)
+        end
         # extend space type name to include the template. Consider this as well for load defs
         space_type.setName("#{space_type.name} - #{template}")
         OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.create_building_system', "Adding loads to space type named #{space_type.name}")
```
change log `BuildingSync.LoadsSystem.add_internal_loads`

TMaile replied:
resolved in https://github.com/BuildingSync/BuildingSync-gem/commit/f38d1d9dbfe61f7b3d8f30fc1fdeb9145f83cf01

corymosiman12 commented on lib/buildingsync/model_articulation/measure.rb:
```diff
@@ -0,0 +1,105 @@
+      read_measure_other_detail(measure_element, ns)
+    end
+
+    def read_measure_other_detail(measure_element, ns)
```
let's define what 'other_details's are.

corymosiman12 commented on lib/buildingsync/model_articulation/wall_system_type.rb:
```diff
@@ -0,0 +1,60 @@
+      end
+    end
+
+    def read(section_element, ns)
```
add `next`?

nllong replied:
empty response should be fine

corymosiman12 commented on lib/buildingsync/model_articulation/facility.rb:
```diff
@@ -39,9 +39,12 @@
 require_relative 'service_hot_water_system'
+require_relative 'measure'
 require 'openstudio/model_articulation/os_lib_model_generation_bricr'
 require 'openstudio/extension/core/os_lib_geometry'
```
just want to make sure - these are being pulled for the OpenStudio::ModelArticulation and OpenStudio::Extension gems respectively, correct?

corymosiman12 commented on lib/buildingsync/model_articulation/facility.rb:
```diff
@@ -39,9 +39,12 @@
+require_relative 'measure'
 require 'openstudio/model_articulation/os_lib_model_generation_bricr'
 require 'openstudio/extension/core/os_lib_geometry'
 require_relative '../helpers/Model.hvac'
```
This seems to be very similar to the `typical_building_from_model` measures...although there's like a million of them in the openstudio-model-articulation-gem and I have no idea how they work (they all seem to be called the same thing...os_lib_model_generation.rb?). We should try to true these up to be inline with one another, although what is used here seems like a good method, since the `model_add_hvac_system` method above is directly from `openstudio-standards`

corymosiman12 replied:
@TMaile can you confirm this will be part of the upcoming scope of work?

corymosiman12 commented on lib/buildingsync/model_articulation/building.rb:
```diff
@@ -125,26 +152,41 @@ def read_built_remodel_year(build_element, ns)
-        major_remodel_year = build_element.elements["#{ns}:YearOfLastMajorRemodel"].text.to_f
-        @built_year = major_remodel_year if major_remodel_year > @built_year
+        @year_major_remodel = build_element.elements["#{ns}:YearOfLastMajorRemodel"].text.to_i
+        @built_year = @year_major_remodel if @year_major_remodel > @built_year
```
Do we want the building year to be superseded by the last major remodel year?  assuming this will affect generation of all constructions, systems, etc.  Is there any other way to do this?

corymosiman12 commented on lib/buildingsync/model_articulation/building.rb:
```diff
@@ -193,17 +249,88 @@ def read_building_form_defaults
     def check_building_faction
```
rename: `check_building_fraction`

TMaile replied:
resolved in https://github.com/BuildingSync/BuildingSync-gem/commit/bf0ef491c32ade5f33a273d95bc3ee0a53f2b44d

corymosiman12 commented on lib/buildingsync/model_articulation/building.rb:
```diff
@@ -193,17 +249,88 @@ def read_building_form_defaults
-        if building_fraction <= 0.0
+        if building_fraction.round(3) < 0.0
+          puts "building fraction is #{ building_fraction }"
           OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Building.check_building_faction', 'Primary Building Type fraction of floor area must be greater than 0. Please lower one or more of the fractions for Building Type B-D.')
```
What are the building types B-D?  can't seem to find definitions.

corymosiman12 commented on lib/buildingsync/model_articulation/building.rb:
```diff
@@ -536,8 +715,42 @@ def set_weather_and_climate_zone_from_epw(climate_zone, epw_file_path, standard_
+    end
 
-      set_climate_zone(climate_zone, standard_to_be_used, stat_file)
+    def get_stat_file(epw_file)
```
What is a STAT file?

corymosiman12 commented on lib/buildingsync/model_articulation/facility.rb:
```diff
@@ -80,21 +126,135 @@ def generate_baseline_osm(epw_file_path, output_path, standard_to_be_used)
+
+      if interval_frequency == 'Month'
+        @interval_reading_monthly.push(MeteredEnergy.new(@energy_resource, interval_frequency, reading_type, interval_reading))
+      elsif interval_frequency == 'Year'
+        @interval_reading_yearly.push(MeteredEnergy.new(@energy_resource, interval_frequency, reading_type, interval_reading))
```
'Year' is invalid.  Should be 'Annual'.  However, we should provide guidance on what this should like like, since we also have under an `auc:ResourceUse/...` the following: 
- `auc:AnnualFuelUseNativeUnits`
- `auc:AnnualFuelUseConsistentUnits`

I would vote for not using an `auc:TimeSeriesData` element for storing annual data, as these are already covered by the above.  Thoughts?

corymosiman12 commented on spec/files/Example – Valid Schema Invalid UseCase.xml:
```diff
@@ -65,8 +65,8 @@
 							<auc:Perimeter>1350</auc:Perimeter>
```
Need to run rubocop on this guy

nllong replied:
yeah add pre-commit?

corymosiman12 commented on spec/files/L000_Instance1.xml:
```diff
@@ -0,0 +1,48 @@
+<auc:BuildingSync xmlns:auc="http://buildingsync.net/schemas/bedes-auc/2019"
```
Need to update with latest file

TMaile replied:
@corymosiman12 where do a get the latest file from?

corymosiman12 replied:
https://github.com/BuildingSync/TestSuite/tree/develop/schematron/v2.2.0/exemplary_files

We still need to change the L000_OpenStudio** files, but basically they should look like that except:
- Not require a blank `auc:Systems` element
- Not require a `auc:Measures`
- Utilize a 'modeled baseline' scenario as described in the scenarios doc.

TMaile replied:
resolved in https://github.com/BuildingSync/BuildingSync-gem/commit/9e242e212afcd6d99349e8c8ab2deb7650db039b

corymosiman12 commented on spec/files/L000_Instance2.xml:
```diff
@@ -0,0 +1,49 @@
+<auc:BuildingSync xmlns:auc="http://buildingsync.net/schemas/bedes-auc/2019"
```
Need to update with latest file

corymosiman12 commented on spec/files/L100_Instance1.xml:
```diff
@@ -0,0 +1,265 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Need to update with latest file

corymosiman12 commented on spec/spec_helper.rb:
```diff
@@ -95,24 +95,23 @@ def run_baseline_simulation(osm_name, epw_name)
       OpenstudioStandards.run_command(cmd)
 
-      expect(File.exist?(osw_file.gsub('in.osw', 'eplusout.sql'))).to be true
+      sql_file = osw_file.gsub('in.osw', 'eplusout.sql')
```
Why do this?

corymosiman12 commented on spec/tests/model_articulation_test/hvac_system_spec.rb:
```diff
@@ -68,4 +68,23 @@
     expect(hvac_system.apply_sizing_and_assumptions(model, output_path, standard, 'Retail', 'PSZ-AC with gas coil heat', '')).to be false
   end
+
+  it 'Should return typical_occupant_usage_value_weeks ' do
```
Seems like this test is mislabeled?

TMaile replied:
changed in https://github.com/BuildingSync/BuildingSync-gem/commit/509cc7bf42619a503fd7ffaa775d76d4f04b53fc

corymosiman12 commented on spec/tests/model_articulation_test/occupancy_types_spec.rb:
```diff
@@ -0,0 +1,123 @@
+    return doc
+  end
+
+  def create_minimum_facility(occupancy_classification, year_of_const, floor_area_type, floor_area_value)
```
This spec contains some really good functions.  We should utilize this in a new `Generator` class - similar to `Translator`, except just generates typical BSync facilities and is able to run them through the workflow.

TMaile replied:
this is a new issue now #72

corymosiman12 commented on lib/buildingsync/model_articulation/site.rb:
```diff
@@ -52,12 +57,28 @@ def initialize(build_element, ns)
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Site.generate_baseline_osm', 'There is no building attached to this site in your BuildingSync file.')
+        raise 'Error: There is no building attached to this site in your BuildingSync file.'
+      elsif number_of_buildings > 1
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Site.generate_baseline_osm', "There is more than one (#{number_of_buildings}) building attached to this site in your BuildingSync file.")
```
Change to: `OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Site.read_xml`...

TMaile replied:
done

corymosiman12 commented on spec/files/building_151.xml:
```diff
@@ -29,7 +29,7 @@
 					<auc:Ownership>Unknown</auc:Ownership>
```
run rubocop

TMaile replied:
replaced tabs with 2 whitespaces

corymosiman12 commented on spec/files/building_151_level1.xml:
```diff
@@ -0,0 +1,1711 @@
+<!-- edited with XMLSpy v2016 rel. 2 sp1 (x64) (http://www.altova.com) by Dan Macumber (NREL) -->
```
run rubocop

TMaile replied:
replaced tabs with 2 whitespaces

corymosiman12 commented on spec/files/building_151_n1.xml:
```diff
@@ -67,8 +67,8 @@
 							<n1:Perimeter>1325</n1:Perimeter>
```
run rubocop

corymosiman12 commented on spec/files/building_151_no_measures.xml:
```diff
@@ -0,0 +1,117 @@
+<!-- edited with XMLSpy v2016 rel. 2 sp1 (x64) (http://www.altova.com) by Dan Macumber (NREL) -->
```
run rubocop

nllong commented on .travis.yml:
```diff
@@ -6,9 +6,13 @@ rvm:
+    # For now use local checkout of model articulation to use the DA branch. This should be removed once
+    # model articulation is released with the bricr model generation file.
+    - env: FAVOR_LOCAL_GEMS=1 OPENSTUDIO_VERSION=2.8.1 OPENSTUDIO_SHA=6914d4f590 RUBYLIB=/usr/local/openstudio-2.8.1/Ruby:/usr/Ruby
+    - env: FAVOR_LOCAL_GEMS=1 OPENSTUDIO_VERSION=2.9.0 OPENSTUDIO_SHA=801faa459c RUBYLIB=/usr/local/openstudio-2.9.0/Ruby:/usr/Ruby
```
We should pin to this to OpenStudio 2.9.1 and 3.0. No action now, just documenting.

TMaile replied:
this is one of the next task after we merge this pull request, see #58

Commit:
update copyrights

Commit:
missed copyrights

Commit:
last copyright, ugh

nllong commented on buildingsync.gemspec:
```diff
@@ -29,7 +29,5 @@ Gem::Specification.new do |spec|
-  spec.add_dependency 'openstudio-extension', '~> 0.1.0'
-  spec.add_dependency 'openstudio-model-articulation', '~> 0.0.1'
-  spec.add_dependency 'openstudio_measure_tester', '~> 0.1.0'
+  spec.add_dependency 'openstudio-model-articulation', '~> 0.1.0'
```
need to update to the latest model articulation. I've asked David to look into emerging in DA's changes to 0.1.x-LTS of openstudio-model-articulation.

TMaile replied:
we will do this after finalising this pull request

nllong commented on lib/buildingsync/model_articulation/spatial_element.rb:
```diff
@@ -42,44 +42,81 @@ module BuildingSync
   class SpatialElement
     def initialize
-      @total_floor_area = nil
+      @total_floor_area = nil # gross floor area
```
not sure this comment adds anything.

TMaile replied:
removed

nllong commented on use_cases/BuildingSync_ASHRAE_Level_1.csv:
```diff
@@ -0,0 +1,3528 @@
+BuildingSyncPath,State,RequiredValues,RequiredPairedElement
```
can we remove this file? This was pre-schematron

TMaile replied:
yes removed

nllong reviewed (approved):
If the tests pass, then I say we merge this down and then branch off to update the other items as needed.

Commit:
update install openstudio script

nllong committed:
fix use case failure for buildingsync 2.0

corymosiman12 commented:
@nllong should we add a `DO NOT MERGE` tag to this (with reasoning), or is this ok to be merged?

nllong commented:
Yes, tag with DO NOT MERGE. DA is going to start working on this again in the coming weeks.

We can't merge because this relies on the parametric schedule measure that has not yet been merged down into OS Model Articulation.

TMaile committed:
change the directory to save weather files into -> "..\lib\data"

TMaile committed:
ensure that reading the first EPW line is successful and if not an error is logged

TMaile committed:
update the error message

TMaile committed:
changed tabs in xml files to 2 whitespaces

TMaile committed:
rubocopy cleanups

TMaile committed:
removed comment

TMaile committed:
removed old BldgSync test file

TMaile committed:
updating to OpenStudio 3.0 and Ruby 2.5, still some errors, but it is a start

TMaile committed:
adjusting travis testing to focus on 3.0.1 only

TMaile committed:
adjusting travis testing to focus on 3.0.1 only

TMaile committed:
adjusting travis testing 2.9

TMaile committed:
update code to use new class OpenStudio::EpwFile

TMaile committed:
including tmp methods

TMaile committed:
update in.osm file to OS 3.0 and adjusted travis again

TMaile committed:
cleanup of required and added dependency on openstudio-standards

corymosiman12 committed:
run all tests
xenial -> bionic
install libstdc++6
./...
add sudo
try dpkg command
update rubylib
add -O flag for deb
use deb file
still linux
linux...
update travis
udpate travis

corymosiman12 committed:
comment out translator_spec, should pass

mrigeshgupta committed:
Code refactore for building gem

TMaile committed:
cleanup of required and added dependency on openstudio-standards

TMaile committed:
fixing issue with reference to OsLib_ModelGeneration

TMaile committed:
Merge pull request #70 from BuildingSync/DA_Update_RefTmp

Refactor code to replace duplicate code in BldgSync gem

TMaile committed:
removed code that was using ol parallel processing of simulations switched to translator.run_osws

TMaile committed:
adding openstudio-ee bldgsync_measures branch

TMaile committed:
Merge pull request #68 from BuildingSync/DA_Update

Update to OpenStudio 3.0 and ruby 2.5
